libs/metrics/base.py

```diff
@@ -0,0 +1,321 @@
+
+    @classmethod
+    def get_entities(cls, config):
+        return config.get_video_sources() if cls.entity == "source" else config.get_areas()
+
+    @classmethod
+    def procces_csv_row(cls, csv_row, object_logs):
+        """
+        Extracts from the `csv_row` the required information to calculate the metric.
+        The extracted information is populated into `object_logs`.
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def generate_hourly_metric_data(cls, object_logs):
+        """
+        Generates the hourly reports for the hours received in `object_logs`.
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def generate_hourly_csv_data(cls, entity_id: str, entity_file: str, time_from: datetime,
+                                 time_until: datetime):
+        if not os.path.isfile(entity_file):
+            entity_type = "Camera" if cls.entity else "Area"
+            logger.warn(f"No data for day! [{entity_type}: {entity_id}]")
+            return
+        objects_logs = {}
+        for hour in range(time_from.hour, time_until.hour):
+            objects_logs[hour] = {}
+        with open(entity_file, newline='') as csvfile:
+            reader = csv.DictReader(csvfile)
+            for row in reader:
+                row_time = datetime.strptime(row["Timestamp"], "%Y-%m-%d %H:%M:%S")
+                if time_from <= row_time < time_until:
+                    cls.procces_csv_row(row, objects_logs)
+            return cls.generate_hourly_metric_data(objects_logs)
+
+    @classmethod
+    def compute_hourly_metrics(cls, config):
+        if not cls.reports_folder:
+            raise Exception(f"The metric {cls} doesn't have configured the folder parameter")
+        base_directory = cls.get_entity_base_directory(config)
+        entities = cls.get_entities(config)
+        current_hour = datetime.now().hour
+        for entity in entities:
+            entity_directory = os.path.join(base_directory, entity["id"])
+            log_directory = None
+            if cls.entity == "source":
+                log_directory = os.path.join(entity_directory, "objects_log")
+            else:
+                # cls.entity == "area"
+                log_directory = os.path.join(entity_directory, "occupancy_log")
+            reports_directory = os.path.join(entity_directory, "reports", cls.reports_folder)
+            # Create missing directories
+            os.makedirs(log_directory, exist_ok=True)
+            os.makedirs(reports_directory, exist_ok=True)
+            if current_hour == 0:
+                # Pending to process the latest hour from yesterday
+                time_until = datetime.combine(date.today(), time(0, 0))
+                report_date = date.today() - timedelta(days=1)
+            else:
+                time_until = datetime.combine(date.today(), time(current_hour, 0))
+                report_date = date.today()
```

Comment:
Why not like this? :thinking: 
```suggestion
            time_until = datetime.combine(date.today(), time(current_hour, 0))
            if current_hour == 0:
                # Pending to process the latest hour from yesterday
                report_date = date.today() - timedelta(days=1)
            else:
                report_date = date.today()
```